src/poisson_numcodecs/poisson.py

```diff
@@ -13,41 +13,115 @@ class Poisson(Codec):
 
     Parameters
     ----------
-    zero_level : float
+    dark_signal : float
         Signal level when no photons are recorded. 
         This should pre-computed or measured directly on the instrument.
     photon_sensitivity : float
         Conversion scalor to convert the measure signal into absolute photon numbers.
         This should pre-computed or measured directly on the instrument.
+    encoded_dtype : str
+        The dtype of the encoded data.
+    decoded_dtype : str
+        The dtype of the decoded data.
+    beta : float
+        The grayscale quantization step expressed in units of noise std dev
+    use_lookup : bool
+        If True, use lookup table to encode and decode the data. 
+        If False, use the forward and inverse functions to encode and decode the data.
     """
     codec_id = "poisson"
 
     def __init__(self, 
-                 zero_level, 
+                 dark_signal, 
                  photon_sensitivity, 
-                 encoded_dtype='int8', 
+                 encoded_dtype='uint8', 
                  decoded_dtype='int16',
+                 beta=0.5,
+                 use_lookup=True
                  ): 
-        self.zero_level = zero_level
+        
+        self.dark_signal = dark_signal
         self.photon_sensitivity = photon_sensitivity
         self.encoded_dtype = encoded_dtype
         self.decoded_dtype = decoded_dtype
+        self.beta = beta
+        self.use_lookup = use_lookup
 
-    def encode(self, buf: np.array) -> bytes:
-        lookup = estimate.make_anscombe_lookup(self.photon_sensitivity)
-        encoded = estimate.lookup(buf, lookup)
-        shape = [encoded.ndim] + list(encoded.shape)
-        shape = np.array(shape, dtype='uint8')
-        return shape.tobytes() + encoded.astype(self.encoded_dtype).tobytes()
-
-    def decode(self, buf: bytes, out=None) -> np.array:
-        lookup = estimate.make_anscombe_lookup(self.photon_sensitivity)
-        inverse = estimate.make_inverse_lookup(lookup)
-        ndims = int(buf[0])
-        shape = [int(_) for _ in buf[1:ndims+1]]
-        arr = np.frombuffer(buf[ndims+1:], dtype=self.encoded_dtype).reshape(shape)
-        decoded = estimate.lookup(arr, inverse)
-        return decoded.astype(self.decoded_dtype)
+        if self.use_lookup:
+            # produce anscombe lookup_tables
+            input_max = np.iinfo(self.decoded_dtype).max
+            xx = (np.r_[:input_max + 1] - self.dark_signal) / self.photon_sensitivity
+
+            # JEROME: I do not understand why we need to subtract np.sqrt(3/8) here
+            # Also, shouldn't te +3/8 be inside the maximum function as xx is a float?
+            forward = 2.0 / self.beta * (np.sqrt(np.maximum(0, xx) + 3/8) - np.sqrt(3/8))
+            
+            # JEROME : I think this might be better syntax?
+            forward = np.round(forward).astype(self.encoded_dtype)
+            self.forward_table = forward
```

Comment:
We subtracted sqrt(⅜) to map zero in the input to zero in the output. We can do the same by using `floor` rather than than `round`. Whether the `+ ⅜` is inside or outside the `np.maximum` then will not make a difference. Since type conversion to int rounds toward zero, we can omit the rounding.
```suggestion
            forward = 2.0 / self.beta * np.sqrt(np.maximum(0, xx + 3/8))            
            self.forward_table = forward.astype(self.encoded_dtype)
```